agent-shell.el

```diff
@@ -91,6 +91,32 @@ Example for devcontainer:
   :type '(repeat string)
   :group 'agent-shell)
 
+(defcustom agent-shell-auto-save-transcript t
```

Comment:
Before opening the feature up for customizations, I'd like to give this feature some time to settle. This gives us time to use it, get a feel for it, and tweak without necessarily breaking users (if we decide to change settings around).

I'd say let's drop `agent-shell-auto-save-transcript` and `agent-shell-transcript-directory` for now. Let's rename `agent-shell-transcript-filename-function` to `agent-shell--transcript-filename-function` (ie. private defvar).

Reply:
> For now if anyone would like to disable things, set agent-shell--transcript-file-path-function to nil.

On second thought, can we make this nil by default and create an internal function we can set it to for the time being? This gives us the opportunity to override ourselves and use it for a bit of time before we role it out to others.